Source/JavaScriptCore/runtime/ISO8601.cpp

```diff
@@ -843,7 +909,38 @@ static std::optional<CalendarRecord> parseCalendar(StringParsingBuffer<Character
     if (*buffer != ']')
         return std::nullopt;
     buffer.advance();
-    return CalendarRecord { WTFMove(result) };
+
+    return CalendarRecord { isCritical, false, WTFMove(result) };
+}
+
+template<typename CharacterType>
+static std::optional<Vector<CalendarRecord>>
+parseCalendar(StringParsingBuffer<CharacterType>& buffer)
+{
+// https://tc39.es/proposal-temporal/#prod-Annotations
+//  Annotations :::
+//      Annotation Annotations[opt]
+
+    Vector<CalendarRecord> result;
+    // https://tc39.es/proposal-temporal/#sec-temporal-parseisodatetime
+    bool calendarWasCritical = false;
+    while (canBeCalendar(buffer)) {
+        auto calendarRecord = parseOneCalendar(buffer);
+        if (!calendarRecord)
+            return std::nullopt;
```

Comment:
Is it right? If we already parsed one calendar for example, and if the second one failed, I don't think returning nullopt here is the right thing.

Reply:
I'm not sure if there's a string that would cause this path to be taken. It seems like `canBeCalendar()` implies that `parseOneCalendar()` will succeed. Maybe it should be an assertion.

Reply:
(well, maybe not. After thinking about it some more, I think `canBeCalendar()` does not imply that `parseOneCalendar()` will succeed. The heuristic in `canBeCalendar()` would return true for a remaining buffer of `[!=..`

Reply:
So then yes, I think this code is correct. A string such as `2025-02-19T13:28[u-ca=gregory][!=..` should return nullopt here, because the `[!=..` is "junk at the end of the string" which is not allowed and should cause ToTemporalPlainDateTime etc. to throw a RangeError.